### PR TITLE
[Regional]Support to add internal application roles for applications.

### PR DESCRIPTION
**Pull request by bhagyasakalanka**
### Proposed changes in this pull request
- When creating set application owners roles as application roles

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java [hunk 1]
```diff
@@ -53,6 +53,7 @@ public class ServiceProvider implements Serializable {
     private static final String IS_MANAGEMENT_APP = "IsManagementApp";
 
     private static final String IS_B2B_SELF_SERVICE_APP = "IsB2BSelfServiceApp";
+    private static final String APPLICATION_ROLE = "applicationRole";
```
Comment (pulasthi7) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java:56]: this requires changing the API contracts, have you done those where applicable? (for SOAP regenerate wsdl/service stub, for rest API resource/version etc). And is this backward compatible?

Comment (pulasthi7) [reply components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java:56]: Also, shouldn't this be an array? supporting multiple roles?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java [hunk 2]
```diff
@@ -446,4 +446,9 @@ public class ApplicationMgtDBQueries {
 
     public static final String GET_TOTAL_SP_CLAIM_USAGES = "SELECT COUNT(*) FROM SP_CLAIM_MAPPING WHERE TENANT_ID = ?" +
             " AND IDP_CLAIM = ?";
+
+    public static final String GET_APPLICATION_ROLES = "SELECT ROLE_NAME FROM SP_APP_ROLE WHERE SP_APP_ID = ?";
```
Comment (pulasthi7) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationMgtDBQueries.java:450]: Is this a new table? have you included the DDLs? also what is the migration impact?

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/provider/ApplicationRoleProvider.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+public interface ApplicationRoleProvider {
+
+    /**
+     * Get user roles.
+     *
+     * @param username Username.
+     * @param tenantId Tenant ID.
+     * @return Array of user roles.
+     */
+    String[] getUserRoles(String username, int tenantId) throws IdentityApplicationManagementException;
```
Comment (pulasthi7) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/provider/ApplicationRoleProvider.java:17]: the method name and the class name is confusing. shouldn't this be just getRoles(), and have application, creator, tenant as the params/context?

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java (hunk 1 above)
Comment (bhagyasakalanka) [reply components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java:56]: This constant is used as the XmlElement name in https://github.com/wso2/carbon-identity-framework/pull/4871/files#diff-1ae98054c1b9ae7db11bb2177188c794ba9b91e42625ff59c59b63c9e0d48124R141. app roles is an array. We can add multiple roles

Comment (bhagyasakalanka) [reply components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java:56]: We are not providing rest API support for this currently.

Comment (bhagyasakalanka) [reply components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java:56]: Add service stubs for app roles

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/provider/ApplicationRoleProvider.java (hunk 3 above)
Comment (bhagyasakalanka) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/provider/ApplicationRoleProvider.java]: Fixed. Change the method signature as above

**Pull request closed without merging**